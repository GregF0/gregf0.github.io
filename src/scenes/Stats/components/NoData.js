import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { useTheme } from '@material-ui/core'
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'
import illustration from './assets/no-data.svg'

export const NoData = () => {
  const theme = useTheme()

  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      alignSelf="center"
      width="90%"
      maxWidth="600px"
      m="auto"
    >
      <Illustration src={illustration} alt="Statistics and charts" />
      <Typography align="center">
        Nothing to show here. Come back after you{' '}
        <LinkStyled to="/timer" color={theme.palette.secondary.main}>
          complete a few sessions
        </LinkStyled>
        .
      </Typography>
    </Box>
  )
}

const Illustration = styled.img`
  display: block;
  width: 90%;
  margin: auto;
`

const LinkStyled = styled(Link)`
  text-decoration: none;
  color: ${({ color }) => color};
`
