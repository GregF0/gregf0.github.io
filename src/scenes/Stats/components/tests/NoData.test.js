import React from 'react'
import { createShallow } from '@material-ui/core/test-utils'
import { NoData } from '../NoData'

describe('<NoData />', () => {
  const shallow = createShallow()
  const wrapper = shallow(<NoData />)

  test('should render <NoData /> correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
