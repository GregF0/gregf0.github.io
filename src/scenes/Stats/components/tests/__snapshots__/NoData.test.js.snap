// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoData /> should render <NoData /> correctly 1`] = `
<Styled(MuiBox)
  alignItems="center"
  alignSelf="center"
  display="flex"
  flexDirection="column"
  justifyContent="center"
  m="auto"
  maxWidth="600px"
  width="90%"
>
  <styled.img
    alt="Statistics and charts"
    src="no-data.svg"
  />
  <WithStyles(ForwardRef(Typography))
    align="center"
  >
    Nothing to show here. Come back after you
     
    <Styled(Link)
      color="#f50057"
      to="/timer"
    >
      complete a few sessions
    </Styled(Link)>
    .
  </WithStyles(ForwardRef(Typography))>
</Styled(MuiBox)>
`;
